### DISCUSSION TITLE
How do I convince a tech lead of using HTMX instead of React?

### DISCUSSION THREAD

**Discussion by albertodiazdorado** [bigskysoftware/htmx#2112] [Q&A]
Let's say we want to write a very simple front-end with very little interactivity and HTMX is a perfect fit. However, my tech lead wants to use React.

Let's say I talk to him and make the case for HTMX. The counter-argument is that HTMX is a niche technology, with uncertain support and that if I ever leave the project he'd rather have React (so that anyone can take over the project) instead of HTMX.

Can I somehow convince him of using HTMX? Or he is right and the non-technical concerns are more important than the technical ones?

**Comment by nickchomey** [comment]
If it's as simple as you say, why not just bang out a mock-up in a day or two and show them? If they aren't interested, you could consider finding a more rational place to work.

Also, I don't see what "support" could possibly be necessary for Htmx... It's dead simple. It's just a tiny library that makes Ajax requests to your backend of choice.

**Reply by albertodiazdorado** [reply]
Thanks for your answer @nickchomey ! Im gonna do that. Is there any particular template engine that you would recommend for Java? Is there any "canonical" way to integrate HTMX with an existing Java backend (that was serving JSON via API until now)?

**Reply by nickchomey** [reply]
I know nothing about Java. Check out the Java channel in the htmx discord - I'm sure people have discussed all of this!

**Reply by linus1412** [reply]
If it is a Spring Boot app using Thymeleaf I'd have a look at [HTMX Spring Boot](https://github.com/wimdeblauwe/htmx-spring-boot)

**Reply by linus1412** [reply]
The author of that library just released a [detailed book](https://leanpub.com/modern-frontends-with-htmx) on the subject.  I'm working my way through it at the moment but it is a good read so far.

**Comment by tim37021** [comment]
I can imagine 10 years later maybe both htmx and React will be dying. If I am the one fresh graduated man just get hired to maintain the legacy code. I would rather like to see it is written in htmx.

**Comment by ivanjeremic** [comment]
The fact that the htmx codebase uses `eval` & the `Function` constructor makes it almost impossible to convince someone. HTMX needs to follow some basic rules first.

**Reply by alexpetros** [reply]
Shame about that. Time to pack it in folks, we had a good run!

**Reply by tim37021** [reply]
There's never a rule. it is useful if you use it wisely, sometimes you just don't want to overkill a problem.

**Reply by alexpetros** [reply]
Rather than just say "basic rules", it's helpful to understand the problem space that `eval` introduces, and how that problem space does and doesn't apply to htmx. I've written a more serious answer about it here: https://github.com/bigskysoftware/htmx/pull/1988#issuecomment-1806290317

**Comment by maltegrosse** [comment]
someone wanna convince him :-)
https://www.youtube.com/watch?v=1xKa4vxon_8&ab_channel=MarkJivko

p.s. never mind, havent seen his 51 subscribers vs 31k htmx github stars :)

**Reply by andryyy** [reply]
Please don’t. I make money fixing six months old, already deprecated, totally bloated security problems people deploy with large frameworks. :)

A classic is "there is no plugin for that" or "it does not work" when someone asks for a small, but off-the-shelf requirement. Like… it’s just four lines of HTML and one more line in the backend.

I say it again: people don’t write backends/frontends anymore, they configure other people’s software. And dependency is **the** killer of applications. HTMX is such a tiny dependency, it will most likely work in 10 years when not updated. I am by no means a JS coder, but it’s just JS. I can always figure it out. It’s not a new language or dialect. Use whatever backend and a core CSS template. Be unique with what you create.

Random example: https://www.stark-sein.eu/ - is this code something you think is superior to, like, anything? Good luck asking anybody changing the slightest function in this mess. Heck, ask the guy who is responsible for this one month later.

There is always "the next big thing". The next framework, the next dependency manager, the next deployment tool. It’s good to get along with these and understanding them. But what’s left of web development? In 10 years we have to buy component blocks to render in a browser from "cloud providers". They are only shown as descriptions. Nobody needs to use their brain anymore.

Look at what’s left of "full stack dev". It used to be _the_ term for big brain time. Not only one dependency, but three for the price of one. Or more. Make it scalable? Oh, I have to ask wether this is possible. It’s "cloud ready". But what the heck is cloud anyway? :)

I don’t want to say new tech is bad (it is not at all!!! I LOVE new tech and use it all the time). But I also love to not add too many third-parties between the tech and me. I want to stay in control as much as possible.

Disclaimer: I have not watched the video, just my take on the context.

**Reply by alexpetros** [reply]
>  And dependency is the killer of applications.

Underappreciated wisdom here.
